File: src/ios/MFPClient.m
/********* MFPClient.m Cordova Plugin Implementation *******/

#import <Cordova/CDV.h>

#import <IMFCore/IMFClient.h>

#import "MFPClient.h"
@implementation MFPClient

- (void)coolMethod:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSString* echo = [command.arguments objectAtIndex:0];
    
    if (echo != nil && [echo length] > 0) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


- (void) initialize:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    
    NSString* route = [command.arguments objectAtIndex:0];
    NSString* guid = [command.arguments objectAtIndex:0];
    NSLog(@"*********** %@" , route);
    
    IMFClient *imfClient = [IMFClient sharedInstance];
    [imfClient initializeWithBackendRoute:route backendGUID:guid];
    
    NSInteger version = [IMFClient  version];
    
    //pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:[NSString stringWithFormat:@"%d", (int)version]];
    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:route];
    //NSLog(@"YO!!! %@",[NSString stringWithFormat:@"%d", (int)version]);
    
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


@end

File: src/ios/MFPClient.swift
//
//  MFPClient.swift
//  HelloCordova
//
//  Created by Larry Nickerson on 9/15/15.
//
//
import Foundation
import IMFCore

@objc(MFPClient) class MFPClient : CDVPlugin {
    
    func initialize(command: CDVInvokedUrlCommand) {
        let route = command.arguments[0] as! String
        let guid = command.arguments[1] as! String
        let client = IMFClient.sharedInstance()
        //TODO handle exception
        client.initializeWithBackendRoute(route, backendGUID: guid)
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAsString: "")
        commandDelegate!.sendPluginResult(pluginResult, callbackId:command.callbackId)
    }
    
    func backendRoute(command: CDVInvokedUrlCommand) {
        let client = IMFClient.sharedInstance()
        let backendRoute: String = client.backendRoute
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAsString: backendRoute)
        commandDelegate!.sendPluginResult(pluginResult, callbackId:command.callbackId)
    }
    
    func backendGUID(command: CDVInvokedUrlCommand) {
        let client = IMFClient.sharedInstance()
        let backendGUID: String = client.backendGUID
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAsString: backendGUID)
        commandDelegate!.sendPluginResult(pluginResult, callbackId:command.callbackId)
    }
}